api/src/main/java/org/openmrs/module/reporting/cohort/definition/library/BuiltInCohortDefinitionLibrary.java

```diff
@@ -125,4 +127,12 @@ public CohortDefinition getDiedDuringPeriod() {
         cd.addParameter(new Parameter("diedOnOrBefore", "reporting.parameter.endDate", Date.class));
         return new MappedParametersCohortDefinition(cd, "diedOnOrAfter", "startDate", "diedOnOrBefore", "endDate");
     }
+
+    @DocumentedDefinition("patientWithAttribute")
+    public CohortDefinition getPatientWithAttribute() {
+        PersonAttributeCohortDefinition cd = new PersonAttributeCohortDefinition();
+        cd.addParameter(new Parameter("attributeType", "reporting.parameter.attributeType", PersonAttributeType.class));
+        cd.addParameter(new Parameter("values", "reporting.parameter.values", String.class, List.class, null));
+        return new MappedParametersCohortDefinition(cd, "attributeType", "attributeType", "values", "values");
```

Comment:
@dkayiwa, this is the way the reporting module works.  If you want to expose any of the configuration properties as parameters, you have do do so explicitly.